code/game/objects/items/devices/swapper.dm

```diff
@@ -0,0 +1,108 @@
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+
+	var/cooldown = 300
+	var/next_use = 0
+	var/obj/item/swapper/linked_swapper
+
+/obj/item/swapper/Destroy()
+	if(linked_swapper)
+		linked_swapper.linked_swapper = null //*inception music*
+		linked_swapper.update_icon()
+		linked_swapper = null
+	return ..()
+
+/obj/item/swapper/update_icon()
+	if(linked_swapper)
+		icon_state = "swapper-linked"
+	else
+		icon_state = "swapper"
+	..()
+
+/obj/item/swapper/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/swapper))
+		var/obj/item/swapper/other_swapper = I
+		if(other_swapper.linked_swapper)
+			to_chat(user, "<span class='warning'>[other_swapper] is already linked. Break the current link to establish a new one.</span>")
+			return
+		if(linked_swapper)
+			to_chat(user, "<span class='warning'>[src] is already linked. Break the current link to establish a new one.</span>")
+			return
+		to_chat(user, "<span class='notice'>You establish a quantum link between the two devices.</span>")
+		linked_swapper = other_swapper
+		other_swapper.linked_swapper = src
+		update_icon()
+		linked_swapper.update_icon()
+	else
+		return ..()
+
+/obj/item/swapper/attack_self(mob/living/user)
+	if(world.time < next_use)
+		to_chat(user, "<span class='warning'>[src] is still recharging.</span>")
+		return
+	if(!linked_swapper)
+		to_chat(user, "<span class='warning'>[src] is not linked with another swapper.</span>")
+		return
+	if(world.time < linked_swapper.cooldown)
+		to_chat(user, "<span class='warning'>[linked_swapper] is still recharging.</span>")
+		return
+	playsound(get_turf(src), 'sound/weapons/flash.ogg', 25, 1)
+	to_chat(user, "<span class='notice'>You activate [src].</span>")
+	next_use = world.time + cooldown //only the one used goes on cooldown
+	addtimer(CALLBACK(src, .proc/swap, user), 25)
+
+/obj/item/swapper/examine(mob/user)
+	. = ..()
+	if(world.time < next_use)
+		to_chat(user, "<span class='warning'>Time left to recharge: [DisplayTimeText(next_use - world.time)]</span>")
+	if(linked_swapper)
+		to_chat(user, "<span class='notice'><b>Linked.</b> Alt-Click to break the quantum link.</span>")
+	else
+		to_chat(user, "<span class='notice'><b>Not Linked.</b> Use on another quantum spin inverter to establish a quantum link.</span>")
+
+/obj/item/swapper/AltClick(mob/living/user)
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
+		return
+	to_chat(user, "<span class='notice'>You break the current quantum link.</span>")
+	linked_swapper.linked_swapper = null
+	linked_swapper.update_icon()
+	linked_swapper = null
+	update_icon()
+
+//Gets the topmost teleportable container
+/obj/item/swapper/proc/get_teleportable_container()
+	var/atom/movable/teleportable = src
+	while(ismovableatom(teleportable.loc))
+		var/atom/movable/AM = teleportable.loc
+		if(AM.anchored)
+			break
+		if(isliving(AM))
+			var/mob/living/L = AM
+			if(L.buckled && L.buckled.anchored)
```

Comment:
Individuals buckled to roller chairs will be pulled, but not their chairs. Intended?

Reply:
not intended, will fix

Reply:
On second thought, it'd require rewriting a lot of the code since it's all based on containers, and it's probably not worth doing. I made it so they're properly unbuckled, though.

Reply:
probably acceptable for it to be strong against wheelchair users